packages/list/index.tsx

```diff
@@ -347,6 +356,24 @@ export default class List<T extends HTMLElement = HTMLElement> extends React.Com
     this.foundation.handleFocusOut(e.nativeEvent, index);
   };
 
+  onDestroy = (index: number) => {
+    const {listItemClassNames} = this.state;
+    delete listItemClassNames[index];
+    this.setState({listItemClassNames});
+  };
+
+  listItemProps = {
+    checkboxList: this.props.checkboxList,
+    radioList: this.props.radioList,
```

Comment:
this won't work for when the user changes these props, right? I think it would be better to place this in some method and use a [memoization helper](https://github.com/alexreardon/memoize-one#readme) to avoid creating this object on every render

Reply:
highly recommended reading [this article](https://overreacted.io/writing-resilient-components/) if you haven't yet :-)

Reply:
ah right - this won't be updated. Is this object created every render? Does that article go over it? I thought having it on the class would make sure its the same object every render.

Reply:
no, the object itself is only created once, but in order to update the props with the user, it should be at least created once per user change. does that make sense?

Reply:
btw is there an ETA for 0.12? I really wanted this component 😢

Reply:
Yup that makes sense. I actually just read that article and it had some great points. Although it made me kinda sad since we follow a lot of the patterns that Dan suggested NOT to do...I think we're aware of them, as others have opened issues against some of these patterns. So we'll have to revisit a lot of these components moving forward.

RE listItemProps: Yes you're right. I think we can use the react.memo like you're saying, and recalculate this in the render.

RE ETA: No official ETA. The rate at which we're going, I don't see a release next week. So I'll update to the Tuesday after 4/30. How does that sound?

Reply:
4/30 sounds good!

about the anti-patterns, we could compile a list of them and open an issue with a checklist, so it's easier to see which ones are missing the update. I would definitely help address them.

Reply:
4/30 is 7 business days...so about 1.5weeks. @TroyTae can you also agree to this date? We can push character count to another release.

Ya definitely... I will compile a list and add it as an issue / let you know.

Reply:
@moog16 
Sorry late response.
I was planning to work on character count in last weekend.
But I had a sick, so I couldn't do nothing.
If you do not mind I want to push back count character release for stable.

Reply:
No need to apologize. That sounds good to me.

Created an issue: https://github.com/material-components/material-components-web-react/issues/817